cpp/src/io/orc/dict_enc.cu

```diff
@@ -130,7 +130,7 @@ __global__ void __launch_bounds__(block_size)
   size_type entry_count{0};
   size_type char_count{0};
   // all threads should loop the same number of times
-  for (auto cur_row = start_row + t; cur_row - t < end_row; cur_row += block_size) {
+  for (int64_t cur_row = start_row + t; cur_row - t < end_row; cur_row += block_size) {
```

Comment:
```suggestion
  for (thread_index_type cur_row = start_row + t; cur_row - t < end_row; cur_row += block_size) {
```
prefer `thread_index_type` over plain integer type. The same to other use cases.